subsys/mgmt/mcumgr/lib/cmd/img_mgmt/include/img_mgmt/img_mgmt.h

```diff
@@ -213,15 +214,14 @@ struct img_mgmt_dfu_callbacks_t {
  * proceeds.  If the callback returns nonzero, the request is rejected with a
  * response containing an `rc` value equal to the return code.
  *
- * @param offset	The offset specified by the incoming request.
- * @param size		The total size of the image being uploaded.
- * @param arg		Optional argument specified when the callback
- *			was configured.
+ * @param req		Image upload request structure
+ * @param action	Image upload action structure
  *
- * @return 0 if the upload request should be accepted; nonzero to reject the request with the
- *	   specified status.
+ * @return	0 if the upload request should be accepted; nonzero to reject
+ *		the request with the specified status.
  */
-typedef int (*img_mgmt_upload_fn)(uint32_t offset, uint32_t size, void *arg);
+typedef int (*img_mgmt_upload_fn)(struct img_mgmt_upload_req req,
```

Comment:
Should we maybe define the `struct img_mgmt_upload_req` as const? Is there any reason we want to let callback modify this thing?

Reply:
We wouldn't made const